ChiModules/LinearBoltzmannSolver/IterativeOperations/lbs_setsource.cc

```diff
@@ -124,43 +122,82 @@ SetSource(LBSGroupset& groupset,
               if ((gprime >= gs_i) and (gprime <= gs_f))
               {
                 double sigma_sm = xs->transfer_matrices[ell].rowI_values[g][t];
-                inscatter_g += sigma_sm * phi_oldp[gprime];
+                inscatter_g += sigma_sm * phi_old_local[ir + gprime];
               }
             }
           }
-          q_mom[g] += inscatter_g;
+          destination_q[ir + g] += inscatter_g;
 
 
           double infission_g = 0.0;
-          const bool fission_avail = ((xs->is_fissile) and (ell == 0));
+          const bool fission_avail = (xs->is_fissile and ell == 0);
 
           //============================== Apply accross-groupset fission
           if (fission_avail and apply_ags_fission_src)
           {
-            //================================ Loop over groups
+            //=============================== Loop over groups
             for (size_t gprime = first_grp; gprime <= last_grp; ++gprime)
             {
               if ((gprime < gs_i) or (gprime > gs_f))
-                infission_g += xs->chi[g] *
-                               xs->nu_sigma_f[gprime] *
-                               phi_oldp[gprime];
+              {
+                // without delayed neutrons
+                if (not options.use_precursors)
+                  infission_g += xs->chi[g] *
+                                 xs->nu_sigma_f[gprime] *
+                                 phi_old_local[ir + gprime];
+
+                // with delayed neutrons
+                else
+                {
+                  //============================== Prompt fission
+                  infission_g += xs->chi_prompt[g] *
+                                 xs->nu_prompt_sigma_f[gprime] *
+                                 phi_old_local[ir + gprime];
+
+                  //============================== Delayed fission
+                  for (size_t j = 0; j < xs->num_precursors; ++j)
+                    infission_g += xs->chi_delayed[g][j] *
+                                   xs->precursor_yield[j] *
+                                   xs->nu_delayed_sigma_f[gprime] *
+                                   phi_old_local[ir + gprime];
+                }
+              }
             }//for gprime
-          }//if zeroth moment
+          }//if fission_avail
 
           //============================== Apply within-groupset fission
           if (fission_avail and apply_wgs_fission_src)
           {
-            //================================ Loop over groups
+            //============================== Loop over groups
             for (size_t gprime = first_grp; gprime <= last_grp; ++gprime)
             {
               if ((gprime >= gs_i) or (gprime <= gs_f))
```

Comment:
Is this possibly an error? Should it not be ```and```?

Reply:
Great catch. Fixed.